src/main/src/main/java/org/geoserver/util/EntityResolverProvider.java

```diff
@@ -26,21 +30,57 @@
 
     public EntityResolverProvider(GeoServer geoServer) {
         this.geoServer = geoServer;
+        this.INTERNAL_ENTITY_RESOLVER = new InternalEntityResolver(geoServer);
     }
 
+    /**
+     * Provide default EntityResolver, used if global settings not provided explicit instructions.
+     *
+     * <p>Primarily used to stage an EntityResolver for test cases.
+     *
+     * @param resolver Entity resolver
+     */
     public static void setEntityResolver(EntityResolver resolver) {
         entityResolver = resolver;
     }
 
+    /**
+     * Obtain EntityResolver respecting GeoServer settings on accessing External Entities.
+     *
+     * @return EntityResolver
+     */
     public EntityResolver getEntityResolver() {
         if (geoServer != null) {
             Boolean externalEntitiesEnabled = geoServer.getGlobal().isXmlExternalEntitiesEnabled();
             if (externalEntitiesEnabled != null && externalEntitiesEnabled) {
                 // XML parser will try to resolve entities
                 return null;
             }
-        }
 
+            if (geoServer.getSettings() != null
+                    && geoServer.getSettings().getProxyBaseUrl() != null) {
+                return INTERNAL_ENTITY_RESOLVER;
```

Comment:
feedback: This should be disabled by default, consider only enabling if environment variable providing list of hosts is known